lib/ntn/ntn.c

```diff
@@ -0,0 +1,265 @@
+
+	req_accuracy = next_req_accuracy;
+	next_req_accuracy = 0;
+
+	if (req_accuracy == 0) {
+		LOG_INF("Location updates not requested");
+	} else {
+		LOG_INF("Location updates requested, accuracy: %d m", req_accuracy);
+
+		evt.modem_location.updates_requested = true;
+		evt.modem_location.accuracy = req_accuracy;
+	}
+
+	if (modem_location_evt_changed(&evt.modem_location)) {
+		event_dispatch(&evt);
+
+		prev_location_update_evt = evt;
+	}
+}
+
+static void at_handler_location(const char *notif)
+{
+	int err;
+	uint32_t next_req_acc_in_s;
+	struct ntn_evt evt = {
+		.type = NTN_EVT_MODEM_LOCATION_UPDATE,
+		.modem_location.updates_requested = false
+	};
+
+	err = parse_location_notif(notif, &req_accuracy, &next_req_accuracy, &next_req_acc_in_s);
+	if (err) {
+		return;
+	}
+
+	if (req_accuracy == 0) {
+		LOG_INF("Location updates not requested immediately");
+
+	} else {
+		LOG_INF("Location updates requested immediately, accuracy: %d m", req_accuracy);
+
+		evt.modem_location.updates_requested = true;
+		evt.modem_location.accuracy = req_accuracy;
+	}
+
+	if (next_req_acc_in_s > 0) {
+		if (next_req_accuracy > 0 && req_accuracy == 0) {
+			/* Location is requested later. The event is sent before the requested time
+			 * to allow enough time for GNSS to get a fix. Modem needs to have
+			 * the current location available at the given time.
+			 */
+			if (next_req_acc_in_s > CONFIG_NTN_MODEM_LOCATION_REQ_MARGIN) {
+				next_req_acc_in_s =
+					next_req_acc_in_s - CONFIG_NTN_MODEM_LOCATION_REQ_MARGIN;
+			} else {
+				next_req_acc_in_s = 0;
+			}
+		}
+
+		k_work_reschedule_for_queue(
+			&ntn_workq,
+			&req_accuracy_update_work,
+			K_SECONDS(next_req_acc_in_s));
+
+		LOG_INF("Location updates requested in %d s, accuracy: %d m",
+			next_req_acc_in_s, next_req_accuracy);
+	} else {
+		k_work_cancel_delayable(&req_accuracy_update_work);
+	}
+
+	if (modem_location_evt_changed(&evt.modem_location)) {
+		event_dispatch(&evt);
+
+		prev_location_update_evt = evt;
+	}
+}
+
+#if defined(CONFIG_UNITY)
+void ntn_on_modem_cfun(int mode, void *ctx)
+#else
+NRF_MODEM_LIB_ON_CFUN(ntn_cfun_hook, ntn_on_modem_cfun, NULL);
+
+static void ntn_on_modem_cfun(int mode, void *ctx)
+#endif /* CONFIG_UNITY */
+{
+	int err;
+
+	ARG_UNUSED(ctx);
+
+	if (mode == LTE_LC_FUNC_MODE_NORMAL || mode == LTE_LC_FUNC_MODE_ACTIVATE_LTE) {
+		/* Subscribe to %LOCATION notifications. */
+		err = nrf_modem_at_printf("AT%%LOCATION=1");
+		if (err) {
+			LOG_WRN("Enabling location notifications failed, error: %d", err);
+			LOG_WRN("Modem firmware with NTN support required");
+		}
+	}
+}
+
+void ntn_register_handler(ntn_evt_handler_t handler)
+{
+	event_handler = handler;
+}
+
+int ntn_modem_location_set(double latitude, double longitude, float altitude, uint32_t validity)
+{
+	int err;
+
+	err = nrf_modem_at_printf(
+		"AT%%LOCATION=2,\"%.6f\",\"%.6f\",\"%.1f\",%d,%d",
+		latitude,
+		longitude,
+		(double)altitude,
+		req_accuracy,
```

Comment:
Is this fine? AT%LOCATION description says this should be estimated accuracy. We are putting the requested accuracy. At least a comment why this is done.

Reply:
Write something along those lines as a comment within the code.